pkg/apis/trainer/v1alpha1/trainjob_types.go

```diff
@@ -131,214 +141,262 @@ type TrainJobSpec struct {
 	// These values can't be set for container with the name: `node`, `dataset-initializer`, or
 	// `model-initializer`. For those containers the envs can only be set via Trainer or Initializer APIs.
 	// +listType=map
 	// +listMapKey=name
+	// +optional
 	Env []corev1.EnvVar `json:"env,omitempty"`
 
-	// Pod volumes to mount into the container's filesystem.
+	// volumeMounts are the volumes to mount into the container's filesystem.
 	// +listType=map
 	// +listMapKey=name
+	// +optional
 	VolumeMounts []corev1.VolumeMount `json:"volumeMounts,omitempty"`
 }
 
 // TrainJobStatus represents the current status of TrainJob.
+// +kubebuilder:validation:MinProperties=1
 type TrainJobStatus struct {
-	// Conditions for the TrainJob.
+	// conditions for the TrainJob.
 	//
 	// +optional
 	// +listType=map
 	// +listMapKey=type
-	// +patchStrategy=merge
-	// +patchMergeKey=type
-	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
 
-	// JobsStatus tracks the child Jobs in TrainJob.
+	// jobsStatus tracks the child Jobs in TrainJob.
 	// +listType=map
 	// +listMapKey=name
+	// +optional
 	JobsStatus []JobStatus `json:"jobsStatus,omitempty"`
 }
 
 type JobStatus struct {
-	// Name of the child Job.
-	Name string `json:"name"`
+	// name of the child Job.
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	Name string `json:"name,omitempty"`
 
-	// Ready is the number of child Jobs where the number of ready pods and completed pods
+	// ready is the number of child Jobs where the number of ready pods and completed pods
 	// is greater than or equal to the total expected pod count for the child Job.
-	Ready int32 `json:"ready"`
+	// +kubebuilder:validation:Minimum=0
+	// +required
+	Ready *int32 `json:"ready,omitempty"`
 
-	// Succeeded is the number of successfully completed child Jobs.
-	Succeeded int32 `json:"succeeded"`
+	// succeeded is the number of successfully completed child Jobs.
+	// +kubebuilder:validation:Minimum=0
+	// +required
+	Succeeded *int32 `json:"succeeded,omitempty"`
 
-	// Failed is the number of failed child Jobs.
-	Failed int32 `json:"failed"`
+	// failed is the number of failed child Jobs.
+	// +kubebuilder:validation:Minimum=0
+	// +required
+	Failed *int32 `json:"failed,omitempty"`
 
-	// Active is the number of child Jobs with at least 1 pod in a running or pending state
+	// active is the number of child Jobs with at least 1 pod in a running or pending state
 	// which are not marked for deletion.
-	Active int32 `json:"active"`
+	// +kubebuilder:validation:Minimum=0
+	// +required
+	Active *int32 `json:"active,omitempty"`
 
-	// Suspended is the number of child Jobs which are in a suspended state.
-	Suspended int32 `json:"suspended"`
+	// suspended is the number of child Jobs which are in a suspended state.
+	// +kubebuilder:validation:Minimum=0
+	// +optional
```

Comment:
It should be required ?
```suggestion
	// +required
```

Reply:
Good catch, thanks. Updated!.